lib/gitsh/completer.rb

```diff
@@ -20,14 +20,27 @@ def initialize(input, line_editor, env, internal_command)
         @internal_command = internal_command
       end
 
-      def complete
-        available_completers.map(&:completions).flatten
+      def call
+        set_completion_append_character
+        matches
       end
 
       private
 
       attr_reader :input, :line_editor, :env, :internal_command
 
+      def set_completion_append_character
+        if matches.size == 1 && matches.first[-1] == '/'
```

Comment:
Extract into a setter method and a `completion_append_character` method; inline the setter method into `call`?